generic_plus/fields.py

```diff
@@ -438,26 +434,11 @@ def __get__(self, instance, instance_type=None):
                 val = getattr(instance, cache_name)
             except AttributeError:
                 db = manager._db or router.db_for_read(rel_model, instance=instance)
-                query = manager.core_filters or {
-                    '%s__pk' % manager.content_type_field_name: manager.content_type.id,
-                    '%s__exact' % manager.object_id_field_name: manager.pk_val,
-                }
-
-                try:
-                    prev_object_id_field = rel_model._meta.get_field(
-                        'prev_%s' % manager.object_id_field_name)
-                except models.FieldDoesNotExist:
-                    pass
-                else:
-                    query['%s__isnull' % prev_object_id_field.attname] = True
-
                 qset = superclass.get_query_set(manager).using(db)
 
                 try:
-                    val = qset.get(**query)
+                    val = qset.get(**manager.core_filters)
```

Comment:
I think `core_filters` gets automatically added in `get_query_set()`, so you may be able to just do `qset.get()`